Make commands from Makefile.base.mk do not work if you don't have nx installed globally in your system

### Describe the bug

Make commands from Makefile.base.mk do not work if you don't have nx installed globally in your system

### Steps to reproduce

1. Go to root
2. Uninstall global nx from the system
3. run `make up`
4. Observe error

### System Info

```shell
Need to install the following packages:
  envinfo@7.10.0
Ok to proceed? (y) 
  System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 86.42 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.15.0/bin/yarn
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
    pnpm: 8.6.1 - ~/Library/pnpm/pnpm
  Browsers:
    Chrome: 114.0.5735.198
    Edge: 114.0.1823.82
    Safari: 16.3
  npmPackages:
    @apollo/client: ^3.7.9 => 3.7.9 
    @apollo/rover: ^0.12.2 => 0.12.2 
    @babel/preset-react: ^7.14.5 => 7.18.6 
    @graphql-typed-document-node/core: ^3.2.0 => 3.2.0 
    @nx/devkit: 16.3.2 => 16.3.2 
    @nx/eslint-plugin: 16.3.2 => 16.3.2 
    @nx/jest: 16.3.2 => 16.3.2 
    @nx/js: 16.3.2 => 16.3.2 
    @nx/linter: 16.3.2 => 16.3.2 
    @nx/node: 16.3.2 => 16.3.2 
    @nx/plugin: 16.3.2 => 16.3.2 
    @nx/react: 16.3.2 => 16.3.2 
    @nx/web: 16.3.2 => 16.3.2 
    @pmmmwh/react-refresh-webpack-plugin: ^0.5.7 => 0.5.10 
    @sb/core: * => 2.0.1 
    @svgr/webpack: ^6.1.2 => 6.5.1 
    @swc-node/register: ^1.4.2 => 1.5.5 
    @swc/cli: 0.1.62 => 0.1.62 
    @swc/core: ^1.2.173 => 1.3.27 
    @swc/helpers: 0.5.1 => 0.5.1 
    @tailwindcss/typography: ^0.5.9 => 0.5.9 
    @testing-library/react: 14.0.0 => 14.0.0 
    @types/gtag.js: ^0.0.12 => 0.0.12 
    @types/jest: 29.4.4 => 29.4.4 
    @types/node: 18.14.2 => 18.14.2 
    @types/react: 18.0.28 => 18.0.28 
    @types/react-dom: 18.0.11 => 18.0.11 
    @types/react-is: 17.0.3 => 17.0.3 
    @types/react-router-dom: 5.3.3 => 5.3.3 
    @types/styled-components: 5.1.26 => 5.1.26 
    @typescript-eslint/eslint-plugin: 5.59.9 => 5.59.9 
    @typescript-eslint/parser: 5.59.9 => 5.59.9 
    babel-jest: 29.4.3 => 29.4.3 
    babel-plugin-relay: ^14.1.0 => 14.1.0 
    babel-plugin-styled-components: 1.10.7 => 1.10.7 
    env-cmd: ^10.1.0 => 10.1.0 
    eslint: ^8.34.0 => 8.34.0 
    eslint-config-prettier: ^8.5.0 => 8.6.0 
    eslint-import-resolver-typescript: ^3.5.3 => 3.5.3 
    eslint-plugin-cypress: ^2.10.3 => 2.12.1 
    eslint-plugin-import: 2.27.5 => 2.27.5 
    eslint-plugin-jsx-a11y: 6.7.1 => 6.7.1 
    eslint-plugin-react: 7.32.2 => 7.32.2 
    eslint-plugin-react-hooks: 4.6.0 => 4.6.0 
    graphql: ^16.6.0 => 16.6.0 
    husky: ^8.0.0 => 8.0.3 
    jest: 29.4.3 => 29.4.3 
    jest-environment-jsdom: 29.4.3 => 29.4.3 
    jsonc-eslint-parser: ^2.1.0 => 2.1.0 
    lint-staged: ^10.5.4 => 10.5.4 
    nx: 16.3.2 => 16.3.2 
    nx-cloud: 16.0.5 => 16.0.5 
    prettier: ^2.6.2 => 2.8.3 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    react-is: 18.2.0 => 18.2.0 
    react-refresh: ^0.10.0 => 0.10.0 
    react-router-dom: 6.11.2 => 6.11.2 
    styled-components: 5.3.6 => 5.3.6 
    tailwindcss: ^3.3.2 => 3.3.2 
    tailwindcss-animate: ^1.0.5 => 1.0.5 
    ts-jest: 29.1.0 => 29.1.0 
    ts-node: 10.9.1 => 10.9.1 
    tsconfig-paths: ^4.1.2 => 4.1.2 
    tslib: ^2.3.0 => 2.4.1 
    typescript: 5.0.4 => 5.0.4 
    url-loader: ^4.1.1 => 4.1.1
```


### Logs

```
make down
nx run --output-style=stream core:docker-compose:down
make: nx: No such file or directory
make: *** [down] Error 1
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/apptension/saas-boilerplate/blob/master/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/apptension/saas-boilerplate/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://docs.demo.saas.apptoku.com/).
- [X] Check that there isn't [already an issue](https://github.com/apptension/saas-boilerplate/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/apptension/saas-boilerplate/discussions) or join our [Discord Chat Server](https://discord.apptension.com).

--- a/.github/workflows/backend.yml
+++ b/.github/workflows/backend.yml
@@ -30,11 +30,11 @@ jobs:
 
       - name: Lint infra
         shell: bash
-        run: node_modules/.bin/nx run backend:lint:js
+        run: pnpm nx run backend:lint:js
 
       - name: Lint & test backend
         shell: bash
-        run: node_modules/.bin/nx run backend:test
+        run: pnpm nx run backend:test
 
       - name: SonarCloud Scan
         uses: sonarsource/sonarcloud-github-action@master

--- a/.github/workflows/docs.yml
+++ b/.github/workflows/docs.yml
@@ -31,8 +31,8 @@ jobs:
 
       - name: Lint
         shell: bash
-        run: node_modules/.bin/nx run docs:lint
+        run: pnpm nx run docs:lint
 
       - name: Build
         shell: bash
-        run: node_modules/.bin/nx run docs:build
+        run: pnpm nx run docs:build

--- a/.github/workflows/infra.yml
+++ b/.github/workflows/infra.yml
@@ -31,8 +31,8 @@ jobs:
 
       - name: "${{ matrix.infra-lib-name }}: Lint"
         shell: bash
-        run: node_modules/.bin/nx run ${{ matrix.infra-lib-name }}:lint
+        run: pnpm nx run ${{ matrix.infra-lib-name }}:lint
 
       - name: "${{ matrix.infra-lib-name }}: Build"
         shell: bash
-        run: node_modules/.bin/nx run ${{ matrix.infra-lib-name }}:build
+        run: pnpm nx run ${{ matrix.infra-lib-name }}:build

--- a/.github/workflows/status-dashboard.yml
+++ b/.github/workflows/status-dashboard.yml
@@ -27,8 +27,8 @@ jobs:
 
       - name: Lint
         shell: bash
-        run: node_modules/.bin/nx run status-dashboard:lint
+        run: pnpm nx run status-dashboard:lint
 
       - name: Build
         shell: bash
-        run: node_modules/.bin/nx run status-dashboard:build
+        run: pnpm nx run status-dashboard:build

--- a/.github/workflows/tools.yml
+++ b/.github/workflows/tools.yml
@@ -27,4 +27,4 @@ jobs:
 
       - name: Lint
         shell: bash
-        run: node_modules/.bin/nx run tools:lint
+        run: pnpm nx run tools:lint

--- a/.github/workflows/webapp.yml
+++ b/.github/workflows/webapp.yml
@@ -27,7 +27,7 @@ jobs:
 
       - name: Build
         shell: bash
-        run: node_modules/.bin/nx run webapp:build
+        run: pnpm nx run webapp:build
 
   test:
     runs-on: ubuntu-20.04
@@ -51,15 +51,15 @@ jobs:
 
       - name: "Lint"
         shell: bash
-        run: node_modules/.bin/nx run webapp:lint
+        run: pnpm nx run webapp:lint
 
       - name: "Type check"
         shell: bash
-        run: node_modules/.bin/nx run webapp:type-check
+        run: pnpm nx run webapp:type-check
 
       - name: "Test"
         shell: bash
-        run: node_modules/.bin/nx run webapp:test --watchAll=false --maxWorkers=20% --coverage
+        run: pnpm nx run webapp:test --watchAll=false --maxWorkers=20% --coverage
 
       - name: SonarCloud Scan
         uses: sonarsource/sonarcloud-github-action@master
@@ -104,15 +104,15 @@ jobs:
 
       - name: "${{ matrix.webapp-lib-name }}: Lint"
         shell: bash
-        run: node_modules/.bin/nx run ${{ matrix.webapp-lib-name }}:lint
+        run: pnpm nx run ${{ matrix.webapp-lib-name }}:lint
 
       - name: "${{ matrix.webapp-lib-name }}: Type check"
         shell: bash
-        run: node_modules/.bin/nx run ${{ matrix.webapp-lib-name }}:type-check
+        run: pnpm nx run ${{ matrix.webapp-lib-name }}:type-check
 
       - name: "${{ matrix.webapp-lib-name }}: Test"
         shell: bash
-        run: node_modules/.bin/nx run ${{ matrix.webapp-lib-name }}:test --watchAll=false --maxWorkers=20% --coverage
+        run: pnpm nx run ${{ matrix.webapp-lib-name }}:test --watchAll=false --maxWorkers=20% --coverage
 
       - name: SonarCloud Scan
         uses: sonarsource/sonarcloud-github-action@master

--- a/.github/workflows/workers.yml
+++ b/.github/workflows/workers.yml
@@ -32,11 +32,11 @@ jobs:
 
       - name: Build emails renderer
         shell: bash
-        run: node_modules/.bin/nx run webapp-emails:build-vite
+        run: pnpm nx run webapp-emails:build-vite
 
       - name: Build
         shell: bash
-        run: node_modules/.bin/nx run workers:build
+        run: pnpm nx run workers:build
 
   test:
     runs-on: ubuntu-20.04
@@ -59,11 +59,11 @@ jobs:
 
       - name: Lint workers
         shell: bash
-        run: node_modules/.bin/nx run workers:lint
+        run: pnpm nx run workers:lint
 
       - name: Test workers
         shell: bash
-        run: node_modules/.bin/nx run workers:test
+        run: pnpm nx run workers:test
 
       - name: SonarCloud Scan
         uses: sonarsource/sonarcloud-github-action@master

--- a/Makefile.base.mk
+++ b/Makefile.base.mk
@@ -34,13 +34,13 @@ export HOST_UID
 USER_SHELL=$(shell env | grep '^SHELL=' | cut -d '=' -f 2)
 
 up:
-	nx run --output-style=stream core:docker-compose:up
+	pnpm nx run --output-style=stream core:docker-compose:up
 
 down:
-	nx run --output-style=stream core:docker-compose:down
+	pnpm nx run --output-style=stream core:docker-compose:down
 
 serve:
-	nx run --output-style=stream core:serve
+	pnpm nx run --output-style=stream core:serve
 
 clean:
 	# remove created images

--- a/bitbucket-pipelines.yml
+++ b/bitbucket-pipelines.yml
@@ -66,9 +66,9 @@ definitions:
         script:
           - *initializeStep
           - *installWebappDeps
-          - node_modules/.bin/nx run webapp:lint
-          - node_modules/.bin/nx run webapp:type-check
-          - node_modules/.bin/nx run webapp:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp:lint
+          - pnpm nx run webapp:type-check
+          - pnpm nx run webapp:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -86,7 +86,7 @@ definitions:
         script:
           - *initializeStep
           - *installWebappDeps
-          - node_modules/.bin/nx run webapp:build
+          - pnpm nx run webapp:build
         caches:
           - pnpmwebapp
           - clis
@@ -100,9 +100,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-core...
-          - node_modules/.bin/nx run webapp-core:lint
-          - node_modules/.bin/nx run webapp-core:type-check
-          - node_modules/.bin/nx run webapp-core:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-core:lint
+          - pnpm nx run webapp-core:type-check
+          - pnpm nx run webapp-core:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -124,9 +124,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-api-client...
-          - node_modules/.bin/nx run webapp-api-client:lint
-          - node_modules/.bin/nx run webapp-api-client:type-check
-          - node_modules/.bin/nx run webapp-api-client:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-api-client:lint
+          - pnpm nx run webapp-api-client:type-check
+          - pnpm nx run webapp-api-client:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -148,9 +148,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-contentful...
-          - node_modules/.bin/nx run webapp-contentful:lint
-          - node_modules/.bin/nx run webapp-contentful:type-check
-          - node_modules/.bin/nx run webapp-contentful:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-contentful:lint
+          - pnpm nx run webapp-contentful:type-check
+          - pnpm nx run webapp-contentful:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -172,9 +172,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-crud-demo...
-          - node_modules/.bin/nx run webapp-crud-demo:lint
-          - node_modules/.bin/nx run webapp-crud-demo:type-check
-          - node_modules/.bin/nx run webapp-crud-demo:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-crud-demo:lint
+          - pnpm nx run webapp-crud-demo:type-check
+          - pnpm nx run webapp-crud-demo:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -196,9 +196,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-documents...
-          - node_modules/.bin/nx run webapp-documents:lint
-          - node_modules/.bin/nx run webapp-documents:type-check
-          - node_modules/.bin/nx run webapp-documents:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-documents:lint
+          - pnpm nx run webapp-documents:type-check
+          - pnpm nx run webapp-documents:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -220,9 +220,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-notifications...
-          - node_modules/.bin/nx run webapp-notifications:lint
-          - node_modules/.bin/nx run webapp-notifications:type-check
-          - node_modules/.bin/nx run webapp-notifications:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-notifications:lint
+          - pnpm nx run webapp-notifications:type-check
+          - pnpm nx run webapp-notifications:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -244,9 +244,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-emails...
-          - node_modules/.bin/nx run webapp-emails:lint
-          - node_modules/.bin/nx run webapp-emails:type-check
-          - node_modules/.bin/nx run webapp-emails:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-emails:lint
+          - pnpm nx run webapp-emails:type-check
+          - pnpm nx run webapp-emails:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -268,9 +268,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-finances...
-          - node_modules/.bin/nx run webapp-finances:lint
-          - node_modules/.bin/nx run webapp-finances:type-check
-          - node_modules/.bin/nx run webapp-finances:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-finances:lint
+          - pnpm nx run webapp-finances:type-check
+          - pnpm nx run webapp-finances:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -292,9 +292,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=webapp-generative-ai...
-          - node_modules/.bin/nx run webapp-generative-ai:lint
-          - node_modules/.bin/nx run webapp-generative-ai:type-check
-          - node_modules/.bin/nx run webapp-generative-ai:test --watchAll=false --maxWorkers=20% --coverage
+          - pnpm nx run webapp-generative-ai:lint
+          - pnpm nx run webapp-generative-ai:type-check
+          - pnpm nx run webapp-generative-ai:test --watchAll=false --maxWorkers=20% --coverage
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -317,9 +317,9 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=backend...
-          - node_modules/.bin/nx run backend:lint:js
+          - pnpm nx run backend:lint:js
           - export DOCKER_BUILDKIT=0  # Workaround for https://community.atlassian.com/t5/Jira-Work-Management-Questions/Can-t-run-compose-in-bitbucket-pipelines-getting-privileged-true/qaq-p/2233411
-          - node_modules/.bin/nx run backend:test
+          - pnpm nx run backend:test
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -347,8 +347,8 @@ definitions:
             --frozen-lockfile
             --filter=workers...
           - export DOCKER_BUILDKIT=0  # Workaround for https://community.atlassian.com/t5/Jira-Work-Management-Questions/Can-t-run-compose-in-bitbucket-pipelines-getting-privileged-true/qaq-p/2233411
-          - node_modules/.bin/nx run workers:lint
-          - node_modules/.bin/nx run workers:test
+          - pnpm nx run workers:lint
+          - pnpm nx run workers:test
           - if [ -z "${SONAR_ORGANIZATION}" ]; then exit 0; fi
           - pipe: sonarsource/sonarcloud-scan:1.4.0
             variables:
@@ -376,8 +376,8 @@ definitions:
             --filter=workers...
             --filter=webapp-emails...
           - export DOCKER_BUILDKIT=0  # Workaround for https://community.atlassian.com/t5/Jira-Work-Management-Questions/Can-t-run-compose-in-bitbucket-pipelines-getting-privileged-true/qaq-p/2233411
-          - node_modules/.bin/nx run webapp-emails:build-vite
-          - node_modules/.bin/nx run workers:build
+          - pnpm nx run webapp-emails:build-vite
+          - pnpm nx run workers:build
         services:
           - docker
         caches:
@@ -397,8 +397,8 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=infra-core...
-          - node_modules/.bin/nx run infra-core:lint
-          - node_modules/.bin/nx run infra-core:build
+          - pnpm nx run infra-core:lint
+          - pnpm nx run infra-core:build
         caches:
           - pnpminfracore
           - clis
@@ -412,8 +412,8 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=infra-shared...
-          - node_modules/.bin/nx run infra-shared:lint
-          - node_modules/.bin/nx run infra-shared:build
+          - pnpm nx run infra-shared:lint
+          - pnpm nx run infra-shared:build
         caches:
           - pnpminfrashared
           - clis
@@ -427,8 +427,8 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=infra-functions...
-          - node_modules/.bin/nx run infra-functions:lint
-          - node_modules/.bin/nx run infra-functions:build
+          - pnpm nx run infra-functions:lint
+          - pnpm nx run infra-functions:build
         caches:
           - pnpminfrafunctions
           - clis
@@ -447,8 +447,8 @@ definitions:
             --filter=docs...
             --filter=backend...
           - export DOCKER_BUILDKIT=0  # Workaround for https://community.atlassian.com/t5/Jira-Work-Management-Questions/Can-t-run-compose-in-bitbucket-pipelines-getting-privileged-true/qaq-p/2233411
-          - node_modules/.bin/nx run docs:lint
-          - node_modules/.bin/nx run docs:build
+          - pnpm nx run docs:lint
+          - pnpm nx run docs:build
         services:
           - docker-small
         caches:
@@ -465,8 +465,8 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=status-dashboard...
-          - node_modules/.bin/nx run status-dashboard:lint
-          - node_modules/.bin/nx run status-dashboard:build
+          - pnpm nx run status-dashboard:lint
+          - pnpm nx run status-dashboard:build
         caches:
           - pnpmstatusdashboard
           - clis
@@ -480,7 +480,7 @@ definitions:
             --include-workspace-root
             --frozen-lockfile
             --filter=tools...
-          - node_modules/.bin/nx run tools:lint
+          - pnpm nx run tools:lint
         caches:
           - pnpmtools
           - clis

